transforms/images/polus-imagej-filter-correlate-plugin/src/main.py

```diff
@@ -0,0 +1,244 @@
+
+        args.append([f for f in _in1.iterdir() if f.is_file()])
+        arg_len = len(args[-1])
+    else:
+        argument_types.append(None)
+        args.append([None])
+
+    # Validate in2
+    in2_types = {
+        "PadAndCorrelateFFT": "RandomAccessibleInterval",
+        "CorrelateFFTC": "RandomAccessibleInterval",
+    }
+
+    # Check that all inputs are specified
+    if _in2 is None and _opName in list(in2_types.keys()):
+        raise ValueError("{} must be defined to run {}.".format("in2", _opName))
+    elif _in2 != None:
+        in2_type = in2_types[_opName]
+
+        # switch to images folder if present
+        if _in2.joinpath("images").is_dir():
+            _in2 = _in2.joinpath("images").absolute()
+
+        args.append([f for f in _in2.iterdir() if f.is_file()])
+        arg_len = len(args[-1])
+    else:
+        argument_types.append(None)
+        args.append([None])
+
+    for i in range(len(args)):
+        if len(args[i]) == 1:
+            args[i] = args[i] * arg_len
+
+    """ Set up the output """
+    out_types = {
+        "PadAndCorrelateFFT": "RandomAccessibleInterval",
+        "CorrelateFFTC": "RandomAccessibleInterval",
+    }
+
+    """ Run the plugin """
+    try:
+        for ind, (
+            in1_path,
+            in2_path,
+        ) in enumerate(zip(*args)):
+            if in1_path != None:
+
+                # Load the first plane of image in in1 collection
+                logger.info("Processing image: {}".format(in1_path))
+                in1_br = BioReader(in1_path)
+
+                # Convert to appropriate numpy array
+                in1 = ij_converter.to_java(
+                    ij, np.squeeze(in1_br[:, :, 0:1, 0, 0]), in1_type
+                )
+                metadata = in1_br.metadata
+                fname = in1_path.name
+                dtype = ij.py.dtype(in1)
+                # Save the shape for out input
+                shape = ij.py.dims(in1)
+            if in2_path != None:
+
+                # Load the first plane of image in in2 collection
+                logger.info("Processing image: {}".format(in2_path))
+                in2_br = BioReader(in2_path)
+
+                # Convert to appropriate numpy array
+                in2 = ij_converter.to_java(
+                    ij, np.squeeze(in2_br[:, :, 0:1, 0, 0]), in2_type
+                )
+
+            # Generate the out input variable if required
+            out_input = ij_converter.to_java(
+                ij, np.zeros(shape=shape, dtype=dtype), "IterableInterval"
+            )
+            
+            # Define the optional border size argument
+            borderSize = shape
+
+            logger.info("Running op...")
+            if _opName == "PadAndCorrelateFFT":
+                out = ij.op().filter().correlate(out_input, in1, in2, borderSize)
+            elif _opName == "CorrelateFFTC":
+                out = ij.op().filter().correlate(out_input, in1, in2, borderSize)
```

Comment:
The only thing that I can glean from the source looking over it for the last ten minutes is that FFTC uses the complex conjugate when applying linear filters. I'm not too worried about it. Let's just leave a note in the README to get a better understanding of how this is working and move on.

Reply:
So what I did here was I included the `borderSize` argument. This will allow the `PadAndCorrelateFFT` method to run properly. I believe there may be a bug somewhere in the ImageJ source code or the ImageJ help menu. The `CorrelateFFTC` method should run by passing just an image and kernel according to ImageJ help. However this method returns indexing error when ImageJ attempts to retrieve elements from the `borderSize` array. Either way this plugin will always run the `PadAndCorrelateFFT` method.

Reply:
I could remove `CorrelateFFTC` method from this plugin since it will never be run. I left it in there for the time being.